lvsfunc/dehalo.py

```diff
@@ -102,3 +103,95 @@ def deemphasize(clip: vs.VideoNode, strength: int = 95,
         .resize.Spline64(clip.width, format=clip.format.id)
     merged = core.std.MaskedMerge(blurred, clip, mask)
     return merged.std.DoubleWeave().std.SelectEvery(2, 0) if interlaced else merged
+
+
+# TO-DO: Add `ref` param that actually works...
+def masked_dha(clip: vs.VideoNode, rx: float = 2.0, ry: float = 2.0,
+               brightstr: float = 1.0, darkstr: float = 0.0,
+               lowsens: float = 50, highsens: float = 50, rfactor: float = 1.0,
+               maskpull: float = 48, maskpush: float = 192, show_mask: bool = False) -> vs.VideoNode:
+    """
+    A combination of the best of DeHalo_alpha and BlindDeHalo3, plus a few minor tweaks to the masking.
+
+    This function is rather sensitive to the rx and ry settings.
+    Set them as low as possible! If the radii are set too high, it will start missing small spots.
+
+    `darkstr` is set to 0 by default in this function. This is because more often than not,
+    it simply does more damage than people will likely want.
+
+    The sensitivity settings are rather difficult to define.
+    In essence, they define the window between how weak an effect is for it to be processed,
+    and how strong it has to be before it's fully discarded.
+
+    Adopted from G41Fun, original by `Orum <https://forum.doom9.org/showthread.php?t=148498>.
+    Heavily modified by LightArrowsEXE.
+
+    :param clip:            Input clip
+    :param rx:              Horizontal radius for halo removal. Must be greater than 1.
+    :param ry:              Vertical radius for halo removal. Must be greater than 1.
+    :param brightstr:       Strength for bright halo removal
+    :param darkstr:         Strength for dark halo removal. Must be between 0 and 1.
+    :param lowsens:         Lower sensitivity range. The lower this is, the more it will process.
+                            Must be between 0 and 100.
+    :param highsens:        Upper sensitivity range. The higher this is, the more it will process.
+                            Must be between 0 and 100.
+    :param rfactor:         Image enlargment factor. Set to >1 to enable some form of aliasing-protection.
+                            Must be greater than 1.
+    :param maskpull:        Mask pulling factor
+    :param maskpush:        Mask pushing factor
+    :param show_mask:       Return mask clip
+
+    :return:                Dehalo'd clip or halo mask clip
+    """
+    if clip.format is None:
+        raise ValueError("masked_dha: 'Variable-format clips not supported'")
+
+    # Original silently changed values around, which I hate. Throwing errors instead.
+    if not all([x >= 1 for x in (rfactor, rx, ry)]):
```

Comment:
`all` accepts generators so you can remove the brackets around the boolean iterator. Additionally, you can convert it to an `any` expression.